frontend/src/components/applyForm/actions.ts

```diff
@@ -1,71 +1,142 @@
 "use server";
 
 import { ErrorObject } from "ajv";
-import { ApiRequestError, parseErrorStatus } from "src/errors";
+import { handleUpdateApplicationForm } from "src/services/fetch/fetchers/applicationFetcher";
 import { getFormDetails } from "src/services/fetch/fetchers/formsFetcher";
+import { ApplicationResponseDetail } from "src/types/applicationResponseTypes";
 import { FormDetail } from "src/types/formResponseTypes";
 
 import { redirect } from "next/navigation";
 
+import { filterFormData } from "./utils";
 import { validateFormData } from "./validate";
 
-type applyFormResponse = {
+type applyFormErrors = {
   errorMessage: string;
   validationErrors: ErrorObject<string, Record<string, unknown>, unknown>[];
+};
+type applyFormResponse = {
+  applicationId: string;
   formData: object;
   formId: string;
+  successMessage: string;
 };
 
 export async function submitApplyForm(
```

Comment:
[blocker] since this is handling both save and submit cases, can we rename this accordingly? maybe `handleFormAction`? hard to come up with a good name for this, which is probably a sign that it should be split up somehow to handle these two cases more discretely. 

at this point I think we should at least update the function name, and make a note to follow up to refactor this into two more clearly deliniated functions that will share most of their behavior through other utilities

Reply:
The save and submit currently do the same thing, submit the data to the same place, the only difference is you get redirected if you click "submit." I updated this, but TBH doesn't seem like a blocking concern for the demo.